7index.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+// task 1:
+
+function detonatorTimer(delay) {
+
+    const interval = setInterval(() => {
+        if (delay > 0) {
+            console.log(delay--);
+        } else {
+            console.log('BOOM!');
+            clearInterval(interval);
+        }
+    }, 1000);
+}
+
+detonatorTimer(3);
+// 3
+// 2
+// 1
+// BOOM!
+
+// task 2:
+
+function detonatorTimer(delay) {
+
+    setTimeout(() => {
+        if (delay === 0) {
+            console.log('BOOM!');
+        } else {
+            console.log(delay)
+            detonatorTimer(delay - 1)
+        }
+    }, 1000);
+}
+
+detonatorTimer(3);
+// 3
+// 2
+// 1
+// BOOM!
+
+// task 3 and 4:
+
+let aboutMe = {
+    name: 'Danya',
+    city: 'Kyiv',
+    gender: 'male',
+    age: 24,
+    hobby: 'aviation',
+    defaultMood: 'hungry',
+    currentMood: 'sleepy',
+    introduce() {
+        console.log(`I am from ${this.city} and my name is ${this.name}`);
+    },
+    prediction() {
+        console.log(`If all goes well, next year I will be ${this.age+1}`);
+    },
+    describeMyMood(){
+        console.log(`I almost always ${this.defaultMood}, but now I'm ${this.currentMood}`);
+    }
+}
+
+let securedSelfIntroduce = Object.freeze(aboutMe).introduce.bind(aboutMe);
```

Comment:
Все що не перезаписується об'являється як константа
```suggestion
const securedSelfIntroduce = Object.freeze(aboutMe).introduce.bind(aboutMe);
```
і далі по коду також

Reply:
В такому випадку нам потрібен тільки bind, для того, щоб не допустити втрати контексту. А якщо нам потрібні убезпечити об'єкт від його змін, достатньо використати freeze один раз